pkg/api/pod/util.go

```diff
@@ -25,59 +25,39 @@ import (
 	"k8s.io/apimachinery/pkg/util/sets"
 	"k8s.io/apimachinery/pkg/util/validation"
 	utilfeature "k8s.io/apiserver/pkg/util/feature"
+	"k8s.io/component-helpers/resource"
 	api "k8s.io/kubernetes/pkg/apis/core"
 	"k8s.io/kubernetes/pkg/apis/core/helper"
 	apivalidation "k8s.io/kubernetes/pkg/apis/core/validation"
 	"k8s.io/kubernetes/pkg/features"
 )
 
-// ContainerType signifies container type
-type ContainerType int
-
-const (
-	// Containers is for normal containers
-	Containers ContainerType = 1 << iota
-	// InitContainers is for init containers
-	InitContainers
-	// EphemeralContainers is for ephemeral containers
-	EphemeralContainers
-)
-
-// AllContainers specifies that all containers be visited
-const AllContainers ContainerType = (InitContainers | Containers | EphemeralContainers)
-
-// AllFeatureEnabledContainers returns a ContainerType mask which includes all container
-// types except for the ones guarded by feature gate.
-func AllFeatureEnabledContainers() ContainerType {
```

Comment:
why was this dropped? this gives callers a way to express "I want all the container types whose features are enabled"

Reply:
We thought this was a legacy function: https://github.com/kubernetes/kubernetes/pull/124609#discussion_r1584798745
Maybe we can add it back when there is a new feature-related container type?

Reply:
no, that will require revisiting all uses of `AllContainers` and trying to decide again whether they care about honoring feature-disablement

Reply:
why not use the `AllContainers` constant directly?

Reply:
AllFeatureEnabledContainers() and AllContainers *happen* to be the same right now, but all callers of AllFeatureEnabledContainers() are intending to ignore any container types disabled by feature gates. Collapsing them to use AllContainers directly loses that information and means any future feature-gates have to try to recover it.